assets/pages/new-contributions-landing/components/ContributionPayment.jsx

```diff
@@ -53,37 +53,42 @@ const mapDispatchToProps = {
 
 // ----- Logic ----- //
 
-function setupPaymentMethod(props: PropTypes): void {
+function initialiseStripeCheckout(props: PropTypes) {
   const {
-    paymentMethod,
     onPaymentAuthorisation,
-    paymentHandler,
     contributionType,
     currency,
     isTestUser,
   } = props;
 
-  if (!paymentHandler[paymentMethod]) {
-    props.isPaymentReady(false);
-
-    switch (paymentMethod) {
-      case 'DirectDebit':
-        // TODO
-        break;
-
-      case 'PayPal':
-        // TODO
-        break;
+  // TODO IMPORTANT: we need to initialise Stripe separately for one-off and recurring
+  // I think this requires us to make
+  // paymentHandlers: { [PaymentMethod]: PaymentHandler | null }
+  // into
+  // paymentHandlers: { [Contrib]: {[PaymentMethod]: PaymentHandler | null }}
+  setupStripeCheckout(onPaymentAuthorisation, contributionType, currency, isTestUser)
```

Comment:
From my testing it looks like this functionality to re-initialise Stripe with a different key was already missing before this PR (since we just checked `paymentHandler[paymentMethod]` which would be the same for one-off, monthly, annual)

Reply:
Yes, have confirmed this by testing. Would be good to see if @regiskuckaertz has any thoughts on this

Reply:
Oh I see, my bad 😵 I guess you have no other choice. Those matrices are coming up in several places now, perhaps it is reasonable to unify them all.

Reply:
as far as the type is concerned, we've managed to unify the matrix! https://github.com/guardian/support-frontend/pull/1019

Reply:
and unlike these object index types, it will enforce that all cases are covered so there shouldn't be runtime errors of trying to call a function which isn't set in the matrix